starter-code/scripts/models/article.js

```diff
@@ -45,18 +46,31 @@ Article.loadAll = function(inputData) {
 /* This function below will retrieve the data from either a local or remote
  source, process it, then hand off control to the View: */
 Article.fetchAll = function() {
-  if (localStorage.hackerIpsum) {
-    /* When our data is already in localStorage:
-    1. We can process and load it,
-    2. Then we can render the index page.  */
-  } else {
-    /* Without our localStorage in memory, we need to:
-    1. Retrieve our JSON file with $.getJSON
-      1.a Load our json data
-      1.b Store that data in localStorage so that we can skip the server call next time,
-      1.c And then render the index page.*/
-  }
-};
+ if (localStorage.hackerIpsum) {
+   /* When our data is already in localStorage:
+   1. We can process and load it,
+   2. Then we can render the index page.  */
+   var hIpsum = JSON.parse(localStorage.getItem('hackerIpsum'));
+   console.log(hIpsum);
+   Article.loadAll(hIpsum);
+   articleView.renderIndexPage();
+
+ } else {
+   /* Without our localStorage in memory, we need to:
+   1. Retrieve our JSON file with $.getJSON
+     1.a Load our json data
+     1.b Store that data in localStorage so that we can skip the server call next time,
+     1.c And then render the index page.*/
+     $.getJSON('../../data/hackerIpsum.json', function(data) {
+       localStorage.setItem('hackerIpsum', JSON.stringify(data));
+       var hIpsum = JSON.parse(localStorage.getItem('hackerIpsum'));
```

Comment:
This is doing a little bit of unnecessary work. It's not breaking anything, but you have retrieved your data from the file already. You have reference to the data you need in memory already as the variable "data". There's no need to retrieve it from local storage in order to pass it to Article.loadAll. Instead, you can save it to local storage and then call Article.loadAll and pass it data directly.